imports/react-ui/settings/knowledgebase/containers/topic/TopicForm.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { compose, gql, graphql } from 'react-apollo';
+import { Loading } from '/imports/react-ui/common';
+import queries from '../../queries';
+import { TopicForm } from '../../components';
+
+const addPropTypes = {
+  getCategoryListQuery: PropTypes.object.isRequired,
+  getBrandListQuery: PropTypes.object.isRequired,
+  save: PropTypes.func.isRequired,
+};
+
+const editPropTypes = {
+  getTopicDetailQuery: PropTypes.object.isRequired,
+  getCategoryListQuery: PropTypes.object.isRequired,
+  getBrandListQuery: PropTypes.object.isRequired,
+  save: PropTypes.func.isRequired,
+};
+
+const TopicEditFormContainer = props => {
+  const { getTopicDetailQuery, getBrandListQuery, getCategoryListQuery } = props;
+
+  if (getTopicDetailQuery.loading || getBrandListQuery.loading || getCategoryListQuery.loading) {
+    return <Loading title="Edit topic" sidebarSize="wide" spin hasRightSideBar />;
+  }
+
+  const updatedProps = {
+    ...props,
+    object: {
+      ...getTopicDetailQuery.knowledgeBaseTopicsDetail,
+      refetch: getTopicDetailQuery.refetch,
+    },
+    brands: getBrandListQuery.brands,
+    categories: getCategoryListQuery.knowledgeBaseCategories,
+  };
+  return <TopicForm {...updatedProps} />;
+};
+
+TopicEditFormContainer.propTypes = editPropTypes;
+
+const TopicAddFormContainer = props => {
+  const { getBrandListQuery, getCategoryListQuery } = props;
+
+  if (getBrandListQuery.loading || getCategoryListQuery.loading) {
+    return <Loading title="Add topic" sidebarSize="wide" spin hasRightSideBar />;
+  }
+
+  const updatedProps = {
+    ...props,
+    brands: getBrandListQuery.brands,
+    categories: getCategoryListQuery.knowledgeBaseCategories,
+  };
+  return <TopicForm {...updatedProps} />;
+};
+
+TopicAddFormContainer.propTypes = addPropTypes;
+
+const topicFormComposer = options => {
+  const { object } = options;
+
+  if (object && object._id) {
+    return compose(
+      graphql(gql(queries.getTopicDetail), {
```

Comment:
Remove this query. We are passing topicObject to form automatically in common Row.js component.